log4j-core/src/main/java/org/apache/logging/log4j/core/async/DisruptorUtil.java

```diff
@@ -60,36 +60,44 @@
     private DisruptorUtil() {
     }
 
-    static long getTimeout(final String propertyName, final long defaultTimeout) {
-        return PropertiesUtil.getProperties().getLongProperty(propertyName, defaultTimeout);
+    static WaitStrategy createWaitStrategy(final String propertyName) {
+        final String strategyStr = PropertiesUtil.getProperties().getStringProperty(propertyName, DisruptorWaitStrategy.TIMEOUT.toString());
+        LOGGER.trace("property {}={}", propertyName, strategyStr);
+        final DisruptorWaitStrategy strategy = DisruptorWaitStrategy.valueOf(Strings.toRootUpperCase(strategyStr));
```

Comment:
I'm not in favor of using an enum and doing a `valueOf` lookup this way. This lookup means that an invalid value (like "INVALID") will result in an `IllegalArgumentException` being thrown. This is not desirable.

In the previous code, an invalid value would result in the `default` case being selected in the `switch` statement, and the logic would correctly fall back to the default `TimeoutBlockingWaitStrategy`.

Reply:
Ok. Changed it back to String